go/ct/common/u256_test.go

```diff
@@ -461,3 +462,213 @@ func TestU256_MarshallingRoundTrip(t *testing.T) {
+			}
+		})
+	}
+}
+
+func TestU256_And(t *testing.T) {
+	maxu64 := uint64(0xffffffffffffffff)
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(1), NewU256(1)},
+		{NewU256(0xff), NewU256(0xff), NewU256(0xff)},
+		{NewU256(maxu64), NewU256(maxu64), NewU256(maxu64)},
+		{NewU256(0x12345678), NewU256(0x87654321), NewU256(0x02244220)},
+		{
+			NewU256(0xffff00, 0xffff00, 0xffff00, 0xffff00),
+			NewU256(0x00ffff, 0x00ffff, 0x00ffff, 0x00ffff),
+			NewU256(0x00ff00, 0x00ff00, 0x00ff00, 0x00ff00),
+		},
+		{
+			NewU256(maxu64, maxu64, maxu64, maxu64),
+			NewU256(maxu64, maxu64, maxu64, maxu64),
+			NewU256(maxu64, maxu64, maxu64, maxu64),
+		},
+		{
+			NewU256(0, maxu64, 0, maxu64),
+			NewU256(0, 0, maxu64, maxu64),
+			NewU256(0, 0, 0, maxu64),
+		},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.a.And(test.b); !want.Eq(got) {
+			t.Errorf("Unexpected result from And, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Or(t *testing.T) {
+	maxu64 := uint64(0xffffffffffffffff)
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(), NewU256(1)},
+		{NewU256(0xff), NewU256(0xff), NewU256(0xff)},
+		{NewU256(maxu64), NewU256(0), NewU256(maxu64)},
+		{NewU256(0x12345678), NewU256(0x87654321), NewU256(0x97755779)},
+		{
+			NewU256(0, maxu64, 0, maxu64),
+			NewU256(0, 0, maxu64, maxu64),
+			NewU256(0, maxu64, maxu64, maxu64),
+		},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.a.Or(test.b); !want.Eq(got) {
+			t.Errorf("Unexpected result from Or, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Xor(t *testing.T) {
+	maxu64 := uint64(0xffffffffffffffff)
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(0), NewU256(1)},
+		{NewU256(0), NewU256(0xff), NewU256(0xff)},
+		{NewU256(maxu64), NewU256(maxu64), U256{}},
+		{NewU256(0x12345678), NewU256(0x87654321), NewU256(0x95511559)},
+		{
+			NewU256(0, maxu64, 0, maxu64),
+			NewU256(0, 0, maxu64, maxu64),
+			NewU256(0, maxu64, maxu64, 0),
+		},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.a.Xor(test.b); !want.Eq(got) {
+			t.Errorf("Unexpected result from Xor, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Srsh(t *testing.T) {
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{NewU256(0x12345678, 0x9abcdef0, 0x12345678, 0x9abcdef0), NewU256(0), NewU256(0x12345678, 0x9abcdef0, 0x12345678, 0x9abcdef0)},
+		{NewU256(0x12345678, 0x9abcdef0, 0x12345678, 0x9abcdef0), NewU256(64), NewU256(0, 0x12345678, 0x9abcdef0, 0x12345678)},
+		{NewU256(0x12345678, 0x9abcdef0, 0x12345678, 0x9abcdef0), NewU256(128), NewU256(0, 0, 0x12345678, 0x9abcdef0)},
+		{NewU256(0x12345678, 0x9abcdef0, 0x12345678, 0x9abcdef0), NewU256(256), NewU256(0, 0, 0, 0)},
```

Comment:
A non-multiple of 64 like 47 would be nice ... to make sure non-word aligned shifts are also covered.